node/bft/src/primary.rs

```diff
@@ -1113,14 +1118,16 @@ impl<N: Network> Primary<N> {
 
     /// Stores the certified batch and broadcasts it to all validators, returning the certificate.
     async fn store_and_broadcast_certificate(&self, proposal: &Proposal<N>, committee: &Committee<N>) -> Result<()> {
-        // Create the batch certificate and transmissions.
-        let (certificate, transmissions) = proposal.to_certificate(committee)?;
-        // Convert the transmissions into a HashMap.
-        // Note: Do not change the `Proposal` to use a HashMap. The ordering there is necessary for safety.
-        let transmissions = transmissions.into_iter().collect::<HashMap<_, _>>();
-        // Store the certified batch.
-        let (storage, certificate_) = (self.storage.clone(), certificate.clone());
-        spawn_blocking!(storage.insert_certificate(certificate_, transmissions))?;
+        let certificate = tokio::task::block_in_place(|| {
```

Comment:
note: this is done via `block_in_place` instead of `spawn_blocking` in order to not have to clone the `Proposal` (which could get expensive enough to defeat the purpose)